### PR TITLE
Fix geo_distance query request generation

### PR DESCRIPTION
**Pull request by TeXitoi**

### CODE REVIEW

File: src/query/geo.rs [hunk 1]
```diff
@@ -162,38 +162,60 @@ impl Serialize for Type {
...
+                          serializer: &mut S) -> Result<(), S::Error>
+        where S: SerializeMap {
+
+        serializer.serialize_entry("distance", &self.distance)?;
+        if let Some(ref t) = self.distance_type {
+            serializer.serialize_entry("distance_type", t)?;
+        }
+        if let Some(ref b) = self.optimize_bbox {
+            serializer.serialize_entry("optimize_bbox", b)?;
+        }
```
Comment (benashford) [inline src/query/geo.rs]: There is a convenience function `serialize_map_optional_kv` in `rs_es::json` that takes the verbosity out of these steps.  However, I like your implementation better, so I'll merge this as-is and tidy up both to do the same thing.

**Comment by benashford** [comment]
Now released as `0.9.1` - https://crates.io/crates/rs-es/0.9.1

**Comment by TeXitoi** [comment]
Thanks.

File: src/query/geo.rs (hunk 1 above)
Comment (TeXitoi) [inline src/query/geo.rs]: I've looked at the GeohashCellQueryOuter implementation that does not use this function. Maybe it also need some cosmetic changes.

Comment (benashford) [inline src/query/geo.rs]: Ah yes, so it does.  That one slipped through the net, I'll tidy that one up while I'm here too.

Thanks 👍

**Pull request merged**